lib/src/lifecycle_module.dart

```diff
@@ -67,9 +67,9 @@ abstract class LifecycleModule {
   /// Calls the onLoad() method, which can be implemented on a Module.
   /// Executes the willLoad and didLoad event streams.
   Future load() async {
-    _willLoadController.add(this);
+    await _willLoadController.add(this);
     await onLoad();
-    _didLoadController.add(this);
+    await _didLoadController.add(this);
```

Comment:
The `await onLoad()` in the previous code is necessary to ensure that the events are properly timed. Not sure why you needed to await the addition of the payload to the event streams. What was breaking without this change?